controller/engine_controller.go

```diff
@@ -972,22 +987,19 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 		}
 		isDRVolume := volume.Status.IsStandby
 		if !isDRVolume {
-			if isReachedLimit, err := m.isReachedConcurrentVolumeBackupRestoreLimit(); err != nil {
-				m.logger.WithError(err).Warn("Failed to check concurrent volume backup restore limit")
-				return nil
-
-			} else if isReachedLimit {
-				m.logger.Debugf("Reached the concurrent volume backup restore limit of %v, retry later", types.SettingNameConcurrentBackupRestorePerNodeLimit)
+			err := m.acquireRestoringCounter(true)
```

Comment:
Why not just add here? no need to change others. You can handle the error returned from the acquire method to decide if need to do defer call.

```go
			m.acquireRestoringCounter(true)
			defer m.acquireRestoringCounter(false)
```

Reply:
We cannot use defer here because restoration could still be in progress when this method returns.

Reply:
@c3y1huang is right because the restore task is asynchronous. I missed this part :+1: